src/analysis/vector/geometry_checker/qgsgeometryoverlapcheck.cpp

```diff
@@ -57,15 +63,15 @@ void QgsGeometryOverlapCheck::collectErrors( const QMap<QString, QgsFeaturePool
       QString errMsg;
       if ( geomEngineA->overlaps( layerFeatureB.geometry().constGet(), &errMsg ) )
       {
-        QgsAbstractGeometry *interGeom = geomEngineA->intersection( layerFeatureB.geometry().constGet() );
+        std::unique_ptr<QgsAbstractGeometry> interGeom( geomEngineA->intersection( layerFeatureB.geometry().constGet() ) );
```

Comment:
There's a lot of places (like this) where QgsGeometryCheckerUtils::createGeomEngine is called and then the resulting engine is used to perform intersects/overlaps checks/etc. But the engine is never prepared, negating most of the benefits of using it. I think QgsGeometryCheckerUtils::createGeomEngine should probably always prepare the created engine before returning it to simplify this -- it'll greatly speed up the whole checking process.

Reply:
Do we *always* want to prepare an engine or are there cases where it's slower? And in the second case, how will a developer using the API know that he has got a case where preparing is worse than not preparing?